nixos/modules/security/duosec.nix

```diff
@@ -195,21 +184,49 @@ in
   };
 
   config = mkIf (cfg.ssh.enable || cfg.pam.enable) {
-     environment.systemPackages = [ pkgs.duo-unix ];
-
-     security.wrappers.login_duo.source = "${pkgs.duo-unix.out}/bin/login_duo";
-     environment.etc = loginCfgFile // pamCfgFile;
-
-     /* If PAM *and* SSH are enabled, then don't do anything special.
-     If PAM isn't used, set the default SSH-only options. */
-     services.openssh.extraConfig = mkIf (cfg.ssh.enable || cfg.pam.enable) (
-     if cfg.pam.enable then "UseDNS no" else ''
-       # Duo Security configuration
-       ForceCommand ${config.security.wrapperDir}/login_duo
-       PermitTunnel no
-       ${optionalString (!cfg.allowTcpForwarding) ''
-         AllowTcpForwarding no
-       ''}
-     '');
+    environment.systemPackages = [ pkgs.duo-unix ];
+
+    security.wrappers.login_duo.source = "${pkgs.duo-unix.out}/bin/login_duo";
+
+    system.activationScripts = {
+      login_duo = mkIf cfg.ssh.enable ''
```

Comment:
What is the expectation of cleanup when using `activationScripts`? ping @Infinisil for opinions.

Reply:
Hmm hard to say, for one I think cleaning up when it's disabled again is needed, but on the other hand having such code run for all disabled modules (which increases in number over time) is kind of unnecessary. Since this handles a secret key I think having it be cleaned up is more important here.

Ideally we might have a `system.deactivationScripts` which are scripts run when a system is deactivated again.

Reply:
@Infinisil `deactivationScripts`: yeah that might be a good idea. What do you think about new variant I pushed for now?

Reply:
@Infinisil reviewing some other modules I'm finding precedent for not cleaning up secrets "tweaked" by `activationScripts`: https://github.com/NixOS/nixpkgs/blob/c2c5dcc00b05f0f6007d7a997dc9d90aefb49f28/nixos/modules/config/ldap.nix#L229

I've gone back to the original way I was doing this as I think that is the most straight forward way. This is the same approach used in the `ldap` module. If you don't see this as a problem - or at least less of a problem than storing secrets in the nix store - I think we should accept the approach I've gone with here.

For context, citing https://duo.com/docs/duounix:
> Treat your secret key like a password
>
> The security of your Duo application is tied to the security of your secret key (skey). Secure it as you would any sensitive credential. Don't share it with unauthorized individuals or email it to anyone under any circumstances!

Reply:
@Infinisil ping because secrets stored in `/etc` with strict permissions are better than secrets stored in `/nix/store`.